sprout/appliances/tasks.py

```diff
@@ -807,11 +817,17 @@ def clone_template_to_appliance__clone_template(self, appliance_id, lease_time_m
                     kwargs['ram'] = appliance.appliance_pool.override_memory
                 if appliance.appliance_pool.override_cpu is not None:
                     kwargs['cpu'] = appliance.appliance_pool.override_cpu
-            appliance.provider_api.deploy_template(
-                appliance.template.name, vm_name=appliance.name,
-                progress_callback=lambda progress: appliance.set_status(
-                    "Deploy progress: {}".format(progress)),
-                **kwargs)
+            if provider_data['type'] == 'openshift' and appliance.template['custom_data']:
+                kwargs['tags'] = appliance.template['custom_data'].get('TAGS')
+            vm_data = appliance.provider_api.deploy_template(appliance.template.name,
+                      vm_name=appliance.name, progress_callback=lambda progress:
+                appliance.set_status("Deploy progress: {}".format(progress)), **kwargs)
+            if provider_data['type'] == 'openshift':
+                with transaction.atomic():
+                    appliance.openshift_ext_ip = vm_data['external_ip']
+                    appliance.openshift_project = vm_data['project']
+                    appliance.ip_address = vm_data['url']
+                    appliance.save()
```

Comment:
While you are here, can you modify the saves to do this?
```python
appliance.save(update_fields=['openshift_ext_ip', 'openshift_project', 'ip_address'])
```

Can you do it for all occurences you are changing or adding? I did not do it originally :)